R/gmailr.R

```diff
@@ -339,7 +339,7 @@ gmailr_query <- function(fun, location, user_id, class = NULL, upload = FALSE,
     stop(cond)
   }
 
-  if (!is.null(class)) {
+  if (!is.null(class) & !is.null(result)) {
```

Comment:
Can you use `&&` instead of `&`. The latter is vectorized, which isn't appropriate here.